bin/run.sh

```diff
@@ -33,14 +33,14 @@ case $1 in
     if [[ $1 == "backend" ]]; then
         shift
         run_back_end_tests
+        coverage report -m
```

Comment:
If you want to send the coverage data from inside the Docker container, you have to pass the various CI related environment variables from the host systems into the container. Codecov has some documentation for [testing with Docker](https://github.com/codecov/support/wiki/Testing-with-Docker).

For ATMO the `bin/test` script is called in the circle.yml which itself uses docker-compose to run the test environment (it uses the docker-compose postgres/redis as well, instead of Circle's instances).

[Here's the code](https://github.com/mozilla/telemetry-analysis-service/blob/684879cac4cbb0212190f976abd8052ec5c18263/bin/test#L11-L17) that fetches a shell script from codecov, which will inspect the host machine's environment and builds a shell variable with all the variables needed for the container codecov call to work successfully.